cmd/tree-walk.go

```diff
@@ -169,11 +169,12 @@ func doTreeWalk(ctx context.Context, bucket, prefixDir, entryPrefixMatch, marker
 	if len(entries) == 0 {
 		return nil
 	}
+
 	for i, entry := range entries {
 		var leaf, leafDir bool
 
 		// Decision to do isLeaf check was pushed from listDir() to here.
-		if delayIsLeaf {
+		if delayIsLeaf && isLeaf != nil {
```

Comment:
Can you also add similar check for leafDir in line 186 ? it makes sense since doTreeWalk is called from many places

Reply:
We can but that function is never called... by healing code.